USWDS-Next: Add contributing.md (#45)
uswds/uswds-next: CONTRIBUTING.md

# Contributing to USWDS web components

Thank you for your interest in contributing to USWDS web components.

> [!Important]
>

> > **This repo is currently in beta and is not yet ready for community code contributions.**

Reviewer: **polish:** Reads a little better.

```suggestion
> **This repo is currently in beta and not yet ready for community code contributions.**
```

Author: Removed the second "is" in this sentence in [03a1c5e](https://github.com/uswds/web-components/pull/45/commits/03a1c5eb80846105a69f858c122c723d1212c67b)

>

> > While the USWDS core team works through big decisions about architecture and style, we ask that you do not submit pull requests. Instead, we encourage you to contribute by your sharing your ideas, concerns, questions, and experiences with the USWDS core team and community.

Author: **note:** I cut out some detail in this version because it felt too verbose, but wanted to add it here as a suggestion in the comments in case there is interest in explaining _why_ we don't want PRs at this time 
 
```suggestion
> While the USWDS core team works through big decisions about architecture and style, we ask that you do not submit pull requests. The code structure and style will continue to evolve in this phase, and we don't want you to spend your time working on a solution that might already be in progress, or solving a problem that no longer exists. Instead, we encourage you to contribute by your sharing your ideas, concerns, questions, and experiences with the USWDS core team and community.
```

Reviewer 2: wondering if we need to say " 'big' decisions about architecture and style". Why not just "decisions about architecture and style"?

Author: Removed the word "big" in [03a1c5e](https://github.com/uswds/web-components/pull/45/commits/03a1c5eb80846105a69f858c122c723d1212c67b)

>

> > Learn more about how you can provide feedback in the [submitting an issue](#submitting-an-issue) and [starting a discussion](#starting-a-discussion) sections below.

Author: **question:** Should we add a line somewhere about how we specifically are looking for users to be beta testers? We can specifically call out that they can help by installing the beta components in their test environments and letting us know how it goes.

Reviewer 2: I like the idea, but will this get lost here? Are we messaging about seeking beta testers in other places?

> > Learn more about how you can provide feedback in the [submitting an issue](#submitting-an-issue) and [starting a discussion](#starting-a-discussion) sections below.

Author: **note:** Wondering if we should specifically call out here that we are interested in users doing beta testing. Maybe something like this?
```suggestion
> We invite you to participate in beta testing by installing our components into your test projects and letting us know how it goes. 
> Learn more about how you can provide feedback in the [submitting an issue](#submitting-an-issue) and [starting a discussion](#starting-a-discussion) sections below.
```

Reviewer 2: Would suggest updating with more active language, but I also have a question: What is it we want to know from them? I'm concerned "how it goes" might be too vague.

## How can I contribute?

### Submitting an issue

> If you have discovered a bug, let us know by [submitting an issue on GitHub](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue). We’ll address it as soon as we are able (but because we are in beta, please be aware that it might be awhile).

Author: **note:** The comment in parentheses is  intended to set expectations about possible response time. There is probably opportunity to improve the language here. Curious to hear thoughts on it.

Reviewer 2: I don't think you need parentheses. What if it's just arranged differently, like:
"If you have discovered a bug, let us know by [submitting an issue on GitHub](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue). We are in beta and may not be able to respond immediately, but we will address as soon as as we are able."

Author: Thanks! Updated to match your suggestion in [03a1c5e](https://github.com/uswds/web-components/pull/45/commits/03a1c5eb80846105a69f858c122c723d1212c67b)

> If you have discovered a bug, let us know by [submitting an issue on GitHub](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue). We’ll address it as soon as we are able (but because we are in beta, please be aware that it might be awhile).

Reviewer: **suggestion:** For convenience you could post the issue link here with a footnote on how to create issue.

Especially since the note at the top has similar language and links to our repo issue template.

```suggestion
If you have discovered a bug, let us know by [submitting an issue on GitHub](https://github.com/uswds/web-components/blob/al/add-contributing.md/CONTRIBUTING.md#submitting-an-issue) [^1]. 

_We’ll address it as soon as we are able (but because we are in beta, please be aware that it might be awhile)._

[^1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue
```

Author: Good notes!

I updated the link locations here to point to our repo rather than "how to" instructions from github.

Also, I ended up opting out of including a footnote because it felt unnecessary to me. The repo links feel pretty intuitive, and the footnotes also felt like they weren't quite right. If we do want to include links to github's "how to" guides, I lean towards making them plain text - something like "Learn more about submitting a GitHub issue..." 

Finally, I removed the on-page anchor links from the "important" note because they did have similar language, and ultimately felt unnecessary. Let me know if you have any objections or other thoughts here.

> If you have discovered a bug or have a new feature to suggest, let us know by [submitting an issue on GitHub](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue). We’ll address it as soon as we are able (but because we are in beta, please be aware that it might be awhile).

Author: ```suggestion
If you have discovered a bug, let us know by [submitting an issue on GitHub](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue). We’ll address it as soon as we are able (but because we are in beta, please be aware that it might be awhile).
```

### Starting a discussion

If you want to share ideas, questions, or concerns with the USWDS core team and community, [open a discussion on GitHub](https://docs.github.com/en/discussions/quickstart#creating-a-new-discussion).

### Submitting a pull request

> At this time, we are not accepting code contributions in the community. We expect to open up community pull request submissions in early 2025, so check back later for instructions on how to contribute pull requests to USWDS web components.

Author: **note:** I took some liberties in assuming a possible PR acceptance timeline here. The goal was to add some clarity and set some expectations (express that eventually we will want contributions - they won't be blocked forever). Please comment if you think this is unnecessary or unadvised.

Reviewer 2: I think it sets expectations, which is good to me.

> At this time, we are not accepting code contributions in the community. We expect to open up community pull request submissions in early 2025, so check back later for instructions on how to contribute pull requests to USWDS web components.

Reviewer: **suggestion:** Emphasis could be useful for scannability.

```suggestion
At this time, we are not accepting <u>code</u> contributions in the community. 

We expect to open up community pull request submissions in early 2025. Check back later for instructions on how to contribute code.
```

Author: I did not include this because the `<u>` element doesn't appear to do anything in the markdown. Additionally, after learning more about the `<u>` element, I lean away from using it because it's styling might cause some users to think it is a link. We could use italics or bold to emphasize the text, but that felt like it created a different tone as I was reading it. Open to thoughts here!

## Code of Conduct

USWDS is committed to building a safe, welcoming, harassment-free culture for everyone. USWDS is a part of the Technology Transformation Services (TTS) within the General Services Administration (GSA) and we expect everyone to follow the GSA TTS Code of Conduct.

> We encourage you to read this project’s Contribution Guide (you are here) and its [README](https://github.com/uswds/uswds-next/blob/develop/README.md). If you want to read more about our open source policy or have questions, check out the [18F Open Source Policy GitHub repository](https://github.com/18f/open-source-policy) or send us an [email](uswds@gsa.gov).

Author: **note:** This is mostly pulled from [uswds' contributing.md file](https://github.com/uswds/uswds/blob/develop/CONTRIBUTING.md#code-of-conduct).
